src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -76,6 +77,8 @@ internal ReturnTypeAndDiagnostics(TypeSymbol returnType, ImmutableArray<Diagnost
                 DeclarationModifiers.Static |
                 syntax.Modifiers.ToDeclarationModifiers();
 
+            this.IsUnsafe = inUnsafeRegion || _declarationModifiers.HasFlag(DeclarationModifiers.Unsafe);
+
```

Comment:
It looks like this is not the right way to address the problem. In my opinion the source of the problem is a special case for local functions in InMethodBionder.GetFlags. We shouldn't have that special case and we definitely shouldn't erase BinderFlags.UnsafeRegion based on the value of IsUnsafe property. By analogy to the regular methods, this property is supposed to reflect modifiers on the local function itself. When the function is marked unsafe itself the BinderFlags.UnsafeRegion flag should be explicitly injected into the binders chain at the appropriate level, see  how WithUnsafeRegionIfNecessary is used for a regular method.